src/tests/unit/mock_storage_recipe_provider.py

```diff
@@ -0,0 +1,97 @@
+from collections import defaultdict
+
+from domain_classes.storage_recipe import StorageAttribute, StorageRecipe
+from enums import StorageDataTypes
+
+all_contained_cases_blueprint = [
+    {
+        "type": "dmss://system/SIMOS/StorageRecipe",
+        "name": "DefaultStorageRecipe",
+        "description": "",
+        "attributes": [
+            {"name": "reference", "type": "dmss://system/SIMOS/Entity", "contained": False},
+            {"name": "references", "type": "dmss://system/SIMOS/Entity", "contained": False},
+        ],
+    }
+]
+
+extended_blueprint = [
+    {
+        "name": "default",
+        "type": "dmss://system/SIMOS/StorageRecipe",
+        "description": "",
+        "storageAffinity": "blob",
+        "attributes": [],
+    }
+]
+
+package_blueprint = [
+    {
+        "type": "dmss://system/SIMOS/StorageRecipe",
+        "name": "DefaultStorageRecipe",
+        "description": "",
+        "attributes": [{"name": "content", "type": "object", "contained": False}],
+    }
+]
+
+uncontained_blueprint = [
+    {
+        "type": "dmss://system/SIMOS/StorageRecipe",
+        "name": "DefaultStorageRecipe",
+        "description": "",
+        "attributes": [
+            {
+                "name": "uncontained_in_every_way",
+                "type": "does_this_matter?",
+                "contained": False,
+                "storageAffinity": "blob",
+            }
+        ],
+    }
+]
+
+blueprint_with_storageAffinity_in_root = [
+    {
+        "name": "default",
+        "type": "dmss://system/SIMOS/StorageRecipe",
+        "description": "",
+        "storageAffinity": "blob",
+        "attributes": [],
+    }
+]
+
+blobContainer = [
+    {
+        "name": "default",
+        "type": "dmss://system/SIMOS/StorageRecipe",
+        "description": "",
+        "attributes": [{"name": "blob", "type": "does_this_matter?", "contained": False}],
+    }
+]
+
+all_storage_recipes = defaultdict(list)
+all_storage_recipes.update(
+    {
+        "all_contained_cases_blueprint": all_contained_cases_blueprint,
+        "ExtendedBlueprint": extended_blueprint,
+        "uncontained_blueprint": uncontained_blueprint,
+        "blob": blueprint_with_storageAffinity_in_root,
+        "blobContainer": blobContainer,
+    }
+)
```

Comment:
Can this be done in one step?
```suggestion
all_storage_recipes: defaultdict(list) =   {
        "all_contained_cases_blueprint": all_contained_cases_blueprint,
        "ExtendedBlueprint": extended_blueprint,
        "uncontained_blueprint": uncontained_blueprint,
        "blob": blueprint_with_storageAffinity_in_root,
        "blobContainer": blobContainer,
    }
```

Reply:
there is a bit inconsistency in naming: camel case vs snake case.

I guess everything should be snake case?

Reply:
No, in your suggestion you are using defaultdict() as a typehint. That will not do the trick.
aye, but this is inherited from mock_blueprint_provider. Don't want to change those types here.